packages/components/addon/components/hds/dialog/index.js

```diff
@@ -0,0 +1,101 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import { getElementId } from '../form/utils/getElementId';
+
+export const DEFAULT_SIZE = 'regular';
+export const DEFAULT_TYPE = 'neutral';
+export const SIZES = ['small', 'regular', 'large'];
+export const TYPES = ['neutral', 'warning', 'critical'];
+
+export default class HdsDialogIndexComponent extends Component {
+  @action
+  setupDialog(element) {
+    this.element = element;
+    // element.showModal();
+  }
+
+  @action
+  onClick(event) {
+    // This may seem counterintuitive, but the <dialog> element is the target only when the backdrop is clicked
+    // otherwise the target would be a child of the <dialog>
+    if (event.target === this.element) {
+      this.closeDialog();
+    }
+  }
+
+  @action
+  closeDialog() {
+    // we call the "onClose" callback if it exists (and is a function)
```

Comment:
Should we consider more granular hooks? eg. `beforeOpen`, `afterOpen`, `beforeClose`, `afterClose`? or maybe just two `onOpen` and `onClose`? thinking out loud here, but eager to hear your opinion on this

(I know this would be the first component for which we would go so granular, but it's also the first component like this)

Reply:
from what I've tested in `cloud-ui` `onOpen` (as in: `afterOpen`) and `onClose` (as in: `beforeClose`) do cover all the current needs. `beforeOpen` and `afterClose` could suggest that the events happen before the component is inserted and after the component is destroyed, respectively, which would be misleading. I would start with these 2 and only expand if needed.

Reply:
> from what I've tested in `cloud-ui` `onOpen` (as in: `afterOpen`) and `onClose` (as in: `beforeClose`) do cover all the current needs. `beforeOpen` and `afterClose` could suggest that the events happen before the component is inserted and after the component is destroyed, respectively, which would be misleading. I would start with these 2 and only expand if needed.

Works for me 👍